src/main/kotlin/com/assisteddarwinism/ultimatesmashtracker/match/MatchController.kt

```diff
@@ -25,17 +26,11 @@ class MatchController {
     lateinit var playerCharacterCombinationRepository: PlayerCharacterCombinationRepository
 
     @GetMapping
-    fun getMatches(@RequestHeader("X-AuthToken") token: String): List<Match> {
+    fun getMatches(@RequestHeader("X-AuthToken") token: String, @RequestParam("playerIds") playerIds: Optional<List<Long>>): List<Match> {
         tokenValidator.checkTokenValid(token)
 
-        return matchRepository.findAll().map { Match(it, playerCharacterCombinationRepository.findAllByMatchId(it.id!!).map { PlayerCharacterRelation(it) }) }
-    }
-
-    @GetMapping("/players/{playerId}")
-    fun getMatches(@RequestHeader("X-AuthToken") token: String, @PathVariable("playerId") playerId: Long): List<Match> {
-        tokenValidator.checkTokenValid(token)
-
-        return getMatches(token).filter { it.players.any { it.playerId == playerId } }
+        var matches = matchRepository.findAll().map { Match(it, playerCharacterCombinationRepository.findAllByMatchId(it.id!!).map { PlayerCharacterRelation(it) }) }
+        return if (playerIds.isPresent) matches.filter { it.players.any { it.playerId in playerIds.get() } } else matches

```

Comment:
I think for clarity maybe an extra line is appropriate.

1: var matches = retrieve from db
2: filter matches by playerids if present
3: return matches

Not a huge problem